### PR TITLE
[Structural] Schemes for OSS and testing

### PR DESCRIPTION
**Pull request by rubenzorrilla**
**📝 Description**
In this PR I'm adding two new static and Bossak schemes to the `StructuralMechanicsApplication`. The requirement of these comes from the need of doing the L2-projections for OSS-stabilised formulations. Nevertheless, I made them with generality in mind so we can eventually use them also for other purposes. Of course these derive from the corresponding `KratosCore` schemes, so they are almost empty.

Changes in the `KratosCore` basically resume to the addition of the reaction variables corresponding to the projections (for the non-linear OSS case) and minors in the base schemes in order to call some methods from the new derived classes.

Tests have been added and modified accordingly.

### CODE REVIEW

**Review by loumalouomega — DISMISSED** [review]

File: applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_implicit_dynamic_solver.py [hunk 1]
```diff
@@ -77,15 +77,26 @@ def _CreateScheme(self):
         process_info[StructuralMechanicsApplication.RAYLEIGH_ALPHA] = self.settings["rayleigh_alpha"].GetDouble()
         process_info[StructuralMechanicsApplication.RAYLEIGH_BETA] = self.settings["rayleigh_beta"].GetDouble()
 
+        # Check that OSS are supported by current scheme
+        oss_available_schemes = ["newmark","bossak"]
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_implicit_dynamic_solver.py]:
```suggestion
        oss_available_schemes = ("newmark","bossak")
```

File: applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_implicit_dynamic_solver.py [hunk 2]
```diff
@@ -77,15 +77,26 @@ def _CreateScheme(self):
...
         # Setting the time integration schemes
-        if(scheme_type == "newmark"):
-            damp_factor_m = 0.0
-            newmark_beta = self.settings["newmark_beta"].GetDouble()
-            mechanical_scheme = KratosMultiphysics.ResidualBasedBossakDisplacementScheme(damp_factor_m, newmark_beta)
-        elif(scheme_type == "bossak"):
-            damp_factor_m = self.settings["damp_factor_m"].GetDouble()
-            newmark_beta = self.settings["newmark_beta"].GetDouble()
-            mechanical_scheme = KratosMultiphysics.ResidualBasedBossakDisplacementScheme(damp_factor_m, newmark_beta)
+        if (scheme_type == "newmark" or scheme_type == "bossak"):
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_implicit_dynamic_solver.py]:
```suggestion
        if scheme_type in ("newmark", "bossak"):
```

**Review by philbucher — CHANGES_REQUESTED** [review]
minor comments, but please reduce the huge test file sizes

**Comment by rubenzorrilla** [comment]
> minor comments, but please reduce the huge test file sizes

The test is the one that we use to have but now with different elements. Note that the large test is in the nightly suite (in the small one I only added a patch test).

File: applications/StructuralMechanicsApplication/tests/cook_membrane_test/cook_membrane_incompressible_oss_results.json
Comment (philbucher) [inline applications/StructuralMechanicsApplication/tests/cook_membrane_test/cook_membrane_incompressible_oss_results.json:1]: I guess you check the results on every node, can you only check a few selected instead?

Comment (rubenzorrilla) [reply applications/StructuralMechanicsApplication/tests/cook_membrane_test/cook_membrane_incompressible_oss_results.json:1]: Sure. Lets do this.

**Comment by rubenzorrilla** [comment]
@philbucher should be ready now. Thanks for noting it.

**Review by philbucher — APPROVED** [review]

**Comment by rubenzorrilla** [comment]
CI is failing because of the MPM application Python tests... @KratosMultiphysics/mpm

**Comment by rubenzorrilla** [comment]
@ncrescenzio MPM application Python tests are still failing. I guess the problem comes from this one
````
2024-07-31T16:32:27.4190338Z test_execution (restart_tests.MPMRestartTestDynamicCantilever2D) ... malloc(): unaligned tcache chunk detected
````

**Comment by ncrescenzio** [comment]
> @ncrescenzio MPM application Python tests are still failing. I guess the problem comes from this one
> 
> ```
> 2024-07-31T16:32:27.4190338Z test_execution (restart_tests.MPMRestartTestDynamicCantilever2D) ... malloc(): unaligned tcache chunk detected
> ```

I have seen, but this is not the same test that was failing two days ago... I am trying to find the problem but at the moment I am not able to reproduce the issue on my laptop...

**Comment by rubenzorrilla** [comment]
Automerge was enabled but I don't really know how this got merged as nothing apparently changed in the branch...

**Comment by ncrescenzio** [comment]
It seems that the test is not always failing (the same happened with the one i fixed in #12593). So I have tried to run again the tests in the CI only for the configuration that is having issues (ubuntu, full debug, clang) and this time they all passed. Same happened in PR #12597.
If in some other PRs the the same test is failing (`MPMRestartTestDynamicCantilever2D`), I suggest to try to run again the tests. At the moment I have no idea how to fix the problem because on my laptop everything is working

**Comment by rubenzorrilla** [comment]
> It seems that the test is not always failing (the same happened with the one i fixed in #12593). So I have tried to run again the tests in the CI only for the configuration that is having issues (ubuntu, full debug, clang) and this time they all passed. Same happened in PR #12597. If in some other PRs the the same test is failing (`MPMRestartTestDynamicCantilever2D`), I suggest to try to run again the tests. At the moment I have no idea how to fix the problem because on my laptop everything is working

According to your description, it could be a memory leak. Did you try to run the case with `FullDebug` build and Valgrind?

**Pull request merged**